VultisigApp/VultisigApp/Chains/Cardano.swift

```diff
@@ -117,8 +117,192 @@ enum CardanoHelper {
                                                                              signatures: allSignatures,
                                                                              publicKeys: publicKeys)
         let output = try CardanoSigningOutput(serializedBytes: compileWithSignature)
-        let result = SignedTransactionResult(rawTransaction: output.encoded.hexString, 
-                                           transactionHash: output.txID.hexString)
+        
+        // Calculate transaction hash manually since WalletCore output.txID is empty for Cardano
+        let transactionHash = calculateCardanoTransactionHash(from: output.encoded)
+        
+        let result = SignedTransactionResult(rawTransaction: output.encoded.hexString,
+                                           transactionHash: transactionHash)
         return result
     }
+    
+    /// Calculate Cardano Transaction ID manually following official specification
+    /// Cardano TX ID = Blake2b-256 hash of the transaction BODY only (not complete transaction)
+    /// Transaction CBOR structure: [body, witness_set, valid_script?, metadata?]
+    private static func calculateCardanoTransactionHash(from transactionData: Data) -> String {
```

Comment:
can you please investigate to see whether there is better ways to get the Cardano transaction hash?  

This code is hard to maintain, if we can find method in walletcore , I would rather we don't write these code